services/api/Tweek.ApiService/Utils/AddonsListExtensions.cs

```diff
@@ -48,20 +48,39 @@ private static IEnumerable<ITweekAddon> GetAddons(IConfiguration configuration)
                     .Select(x => Assembly.CreateQualifiedName(x["AssemblyName"], x["ClassName"]))
             );
 
+            _addonsCache = GetSelectedAddons(selectedAddons);
+
+            return _addonsCache;
+        }
+
+        public static IEnumerable<ITweekAddon> GetSelectedAddons(HashSet<string> selectedAddons)
+        {
+            var assemblies = GetRuntimeAssemblies();
+
+            var addonTypes = GetAddonTypeFromAssemblies(assemblies);
+
+            return InstantiateSelectedAddonTypes(addonTypes, selectedAddons);
+        }
+
+        public static IEnumerable<Assembly> GetRuntimeAssemblies()
```

Comment:
can be private, not sure if there's a value for splitting it to different method in the correct case